curated_transformers/tokenization/tokenizer.py

```diff
@@ -218,32 +295,82 @@ def _encode_chunks(self, input: Iterable[InputChunks]) -> PiecesWithIds:
 
         return PiecesWithIds(ids=ids, pieces=pieces)
 
+    @property
+    def eos_piece(self) -> Optional[str]:
+        return self._eos_piece
+
     @classmethod
-    def from_file(cls: Type[Self], path: Path) -> Self:
+    def from_dir(cls: Type[Self], path: Path) -> Self:
         """
-        Load the tokenizer from a ``tokenizer.json`` file.
+        Load the tokenizer from a directory with a ``tokenizer.json`` file.
 
         :param path:
             Path to the tokenizer file.
         """
-        hf_tokenizer = HFTokenizer.from_file(str(path))
-        return cls(hf_tokenizer)
+        tokenizer_path = path / TOKENIZER_JSON
+        config_path = path / HF_TOKENIZER_CONFIG
+        special_tokens_map_path = path / SPECIAL_TOKENS_MAP
+        hf_tokenizer = HFTokenizer.from_file(str(tokenizer_path))
+        config = None
+        if config_path.is_file():
+            with open(config_path, encoding="utf-8") as f:
+                config = json.load(f)
+        special_tokens_map = None
+        if special_tokens_map_path.is_file():
+            with open(special_tokens_map_path, encoding="utf-8") as f:
+                special_tokens_map = json.load(f)
+        return cls(
+            tokenizer=hf_tokenizer,
+            config=config,
+            special_tokens_map=special_tokens_map,
+        )
 
     @classmethod
-    def from_hf_hub(cls: Type[Self], *, name, revision="main") -> Self:
+    def from_hf_hub(cls: Type[Self], *, name: str, revision: str = "main") -> Self:
         hf_tokenizer = HFTokenizer.from_pretrained(name, revision)
-        return cls(hf_tokenizer)
+
+        try:
+            config = get_tokenizer_config(name=name, revision=revision)
+        except EntryNotFoundError:
+            config = None
```

Comment:
Can we modify `get_tokenizer_config`/`get_special_tokens_map` to return a `None` instead of throwing?